addons/binding/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/protocol/KodiConnection.java

```diff
@@ -332,6 +328,41 @@ private String convertToText(JsonElement element) {
         // }
     }
 
+    private String convertToImageUrl(JsonElement element) {
+        String text = convertToText(element);
+        if (!text.isEmpty()) {
+            try {
+                // we have to strip ending "/" here because Kodi returns a not valid path and filename
+                // "fanart":"image://http%3a%2f%2fthetvdb.com%2fbanners%2ffanart%2foriginal%2f263365-31.jpg/"
+                // "thumbnail":"image://http%3a%2f%2fthetvdb.com%2fbanners%2fepisodes%2f263365%2f5640869.jpg/"
+                String encodedURL = URLEncoder.encode(StringUtils.stripEnd(text, "/"), "UTF-8");
+                return imageUri.resolve(encodedURL).toString();
+            } catch (UnsupportedEncodingException e) {
+                return text;
+            }
+        }
+        return text;
```

Comment:
I doubt whether this is a proper fallback? WDYT?

Reply:
I am going to return `null` here.